tests/test_DG_advection.py

```diff
@@ -0,0 +1,107 @@
+from dcore import *
+from firedrake import IcosahedralSphereMesh, Expression, SpatialCoordinate, \
+    as_vector, VectorFunctionSpace, File
+import itertools
+from math import pi
+
+
+def setup_DGadvection(dirname, vector=False):
+
+    refinements = 3  # number of horizontal cells = 20*(4^refinements)
+    R = 1.
+    dt = pi/3*0.001
+
+    mesh = IcosahedralSphereMesh(radius=R,
+                                 refinement_level=refinements)
+    global_normal = Expression(("x[0]/sqrt(x[0]*x[0]+x[1]*x[1]+x[2]*x[2])",
+                                "x[1]/sqrt(x[0]*x[0]+x[1]*x[1]+x[2]*x[2])",
+                                "x[2]/sqrt(x[0]*x[0]+x[1]*x[1]+x[2]*x[2])"))
+    mesh.init_cell_orientations(global_normal)
```

Comment:
Heh, oops.


________________________________
From: Lawrence Mitchell <notifications@github.com>
Sent: 12 April 2016 18:02
To: firedrakeproject/dcore
Cc: Shipton, Jemma
Subject: Re: [firedrakeproject/dcore] Dg advection (#23)


In tests/test_DG_advection.py<https://github.com/firedrakeproject/dcore/pull/23#discussion_r59412668>:

> +import itertools
> +from math import pi
> +
> +
> +def setup_DGadvection(dirname, vector=False):
> +
> +    refinements = 3  # number of horizontal cells = 20*(4^refinements)
> +    R = 1.
> +    dt = pi/3*0.001
> +
> +    mesh = IcosahedralSphereMesh(radius=R,
> +                                 refinement_level=refinements)
> +    global_normal = Expression(("x[0]/sqrt(x[0]*x[0]+x[1]*x[1]+x[2]*x[2])",
> +                                "x[1]/sqrt(x[0]*x[0]+x[1]*x[1]+x[2]*x[2])",
> +                                "x[2]/sqrt(x[0]*x[0]+x[1]*x[1]+x[2]*x[2])"))
> +    mesh.init_cell_orientations(global_normal)


You know the normal doesn't have to be normalised, right?

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/firedrakeproject/dcore/pull/23/files/f322e474cebe58e9debfb0d41281be1f458c7392#r59412668>